core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,46 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class JavaGrepLambdaImp extends JavaGrepImp {
+
+    public static void main(String[] args) {
+        if (args.length != 3) {
+            throw new IllegalArgumentException("Must provide: regex rootPath outFile");
+        }
+
+        JavaGrepLambdaImp javaGrepLambdaImp = new JavaGrepLambdaImp();
+        javaGrepLambdaImp.setRegex(args[0]);
+        javaGrepLambdaImp.setRootPath(args[1]);
+        javaGrepLambdaImp.setOutFile(args[2]);
+
+        try {
+            javaGrepLambdaImp.process();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
never use printstacktrack as it will only print to stderr or stdout

user `logger.error("your message", e)`